### PR TITLE
Updating FastjetJetProducer with fillDescriptions

### PR DESCRIPTION
**Pull request by alefisico**
Update needed for using grooming techniques at HLT (previously incompatible with confdb). A brief summary of this PR:
- Update FastjetJetProducer, and the classes affected by this change, with fillDescriptions method.
- It also removes PFJetParameters_cfi and AnomalousCellParameters_cfi dependencies in the python/*py config files not longer needed. (Default parameters in fillDescriptions part)

This is a first PR "updating and cleaning" this package as discussed with @rappoccio @ahinzmann @jdolen @nhanvtran @gkasieczka. I might submit another one just cleaning all the python files no longer needed.

### CODE REVIEW

**Comment by perrotta** [comment]
please test

**Comment by slava77** [comment]
@alefisico 
please provide a link to slides outlining the changes made in this PR.
This change was not mentioned in the JME status and plans presentation on Apr 21. So, it's probably something new.

The PR title does no represent the changes made. Please edit the title.

**Comment by slava77** [comment]
@alefisico 
before we restart the tests please confirm that the standard check ```runTheMatrix.py -l limited -i all ``` works

**Comment by alefisico** [comment]
@slava77 the title does represent the changes made. 
The main (and needed) change is in the FastjetJetProducer class, but this is a "subclass" of the VirtualJetProducer, therefore I needed to change VirtualJetProducer. But there are some other classes that depends on that VirtualJetProducer, so I modified it accordingly. 
Then, several python config files uses the PFJetParameter_cfi files which had the default parameters of the Virtual and FastjetJetproducer, so with the change is not longer needed. 
But if you still think that the title should change, please let me know your suggestion.
About the runTheMatrix, I am rerunnning it again just in case.

**Comment by slava77** [comment]
@alefisico 
OK, maybe after slides are available with the details and the code inspection it becomes more clear.
"Updating xyz with fillDescriptions" implies rather technical change that affects configuration of modules and has no change in functionality (neither addition nor removal).

**Comment by ahinzmann** [comment]
@slava77 Just to confirm, this is supposed to be a pure technical change, adding fillDescriptions, but no added/changed/removed functionality.

**Comment by alefisico** [comment]
yes @ahinzmann @slava77 this is a pure technical change, it should be totally transparent with the rest of the code. As pointed out by @ahinzmann there is no added/changed/removed functionality in the FastjetJetProducer or any "subprocesses".

File: RecoJets/JetProducers/interface/FastjetJetProducer.h [hunk 1]
```diff
@@ -0,0 +1,118 @@
+#ifndef FastjetJetProducer_h
+#define FastjetJetProducer_h
```
Comment (perrotta) [inline RecoJets/JetProducers/interface/FastjetJetProducer.h]: Please use an include guard matching the full file name subsystem_package_fileName_h

File: RecoJets/JetProducers/plugins/SubEventGenJetProducer.h [hunk 2]
```diff
@@ -11,7 +11,7 @@
  ************************************************************/
```
Comment (perrotta) [inline RecoJets/JetProducers/plugins/SubEventGenJetProducer.h]: This is a duplicate of  RecoJets/JetProducers/interface/SubEventGenJetProducer.h

File: RecoJets/JetProducers/interface/FastjetJetProducer.h (hunk 1 above)
Comment (perrotta) [inline RecoJets/JetProducers/interface/FastjetJetProducer.h]: Please use an include guard matching the full file name subsystem_package_fileName_h

File: RecoJets/JetProducers/interface/VirtualJetProducer.h [hunk 3]
```diff
@@ -0,0 +1,223 @@
+#ifndef newVirtualJetProducer_h
+#define newVirtualJetProducer_h
```
Comment (perrotta) [inline RecoJets/JetProducers/interface/VirtualJetProducer.h]: Please use an include guard matching the full file name subsystem_package_fileName_h

File: RecoJets/JetProducers/plugins/FastjetJetProducer.cc [hunk 4]
```diff
@@ -48,483 +66,462 @@
...
+	yCut_ = iConfig.getParameter<double>("yCut");
+
+	useFiltering_ = iConfig.getParameter<bool>("useFiltering");
+	rFilt_ = iConfig.getParameter<double>("rFilt");
+	nFilt_ = iConfig.getParameter<int>("nFilt");
+	useDynamicFiltering_ = iConfig.getParameter<bool>("useDynamicFiltering");
+	rFiltFactor_ = iConfig.getParameter<double>("rFiltFactor");
+
+	useTrimming_ = iConfig.getParameter<bool>("useTrimming");
+	//rFilt_ = iConfig.getParameter<double>("rFilt"); 	//// duplicate parameter
```
Comment (perrotta) [inline RecoJets/JetProducers/plugins/FastjetJetProducer.cc]: Remove this commented out line

**Review by perrotta — COMMENTED** [review]
Not finished with review yet...

File: RecoJets/JetProducers/plugins/VirtualJetProducer.cc [hunk 5]
```diff
@@ -116,178 +119,128 @@ void VirtualJetProducer::makeProduces( std::string alias, std::string tag )
...
+	makeProduces( moduleLabel_, jetCollInstanceName_ );
+	produces<vector<double> >("rhos");
+	produces<vector<double> >("sigmas");
+	produces<double>("rho");
+	produces<double>("sigma");
+
+	/*input_candidateview_token_ = consumes<reco::CandidateView>(src_);
+	input_vertex_token_ = consumes<reco::VertexCollection>(srcPVs_);
+	input_candidatefwdptr_token_ = consumes<vector<edm::FwdPtr<reco::PFCandidate> > >(src_);
+	input_packedcandidatefwdptr_token_ = consumes<vector<edm::FwdPtr<pat::PackedCandidate> > >(src_);*/
```
Comment (perrotta) [inline RecoJets/JetProducers/plugins/VirtualJetProducer.cc]: Remove the commented out lines (they are already there at lines 156-159

**Review by perrotta — COMMENTED** [review]
Finished with my review.
Please, either apply the fixes or answer why you should better not: then tests will be rerun

**Comment by alefisico** [comment]
@perrotta thanks for the review, but it seems that some JMAR people have some concerns about the PR. I will close this PR for now, discussed with them internally and then redo the PR.

**Pull request closed without merging**